### ISSUE TITLE
Futter Examples dependency conflicts

### ISSUE THREAD

**Issue by LaughingCabbage** [flutter/flutter#18624]
## Steps to Reproduce
```
cd flutter/examples/catalog
flutter update-packages --force-upgrade
flutter run
```

Output tells me I need an older version of dependency X. This continues as I downgrade the versions in `pubspec.yaml`. 

## Logs
```
C:\Users\Kevin\Documents\flutter\examples\catalog>flutter update-packages --force-upgrade
Upgrading packages...
WARNING: the following packages use hard-coded version constraints:
  - mockito: 3.0.0-alpha+5
Running "flutter packages upgrade" in flutter_update_packages_8c8e61a9-743b-11e8-9271-309c23856a70...  4.3s
Running "flutter packages get" in automated_tests...         0.5s
Running "flutter packages get" in complex_layout...          0.6s
Running "flutter packages get" in microbenchmarks...         0.6s
Running "flutter packages get" in bots...                    0.5s
Running "flutter packages get" in devicelab...               0.5s
Running "flutter packages get" in channels...                0.5s
Running "flutter packages get" in external_ui...             0.5s
Running "flutter packages get" in flavors...                 0.5s
Running "flutter packages get" in platform_interaction...    0.5s
Running "flutter packages get" in ui...                      0.6s
Running "flutter packages get" in manual_tests...            0.5s
Running "flutter packages get" in vitool...                  0.5s
Running "flutter packages get" in tools...                   0.5s
Running "flutter packages get" in catalog...                 0.5s
Running "flutter packages get" in flutter_gallery...         0.6s
Running "flutter packages get" in flutter_view...            0.3s
Running "flutter packages get" in hello_world...             0.5s
Running "flutter packages get" in layers...                  0.5s
Running "flutter packages get" in platform_channel...        0.5s
Running "flutter packages get" in platform_channel_swift...  0.5s
Running "flutter packages get" in platform_view...           0.3s
Running "flutter packages get" in stocks...                  0.5s
Running "flutter packages get" in flutter...                 0.5s
Running "flutter packages get" in flutter_driver...          0.5s
Running "flutter packages get" in flutter_goldens...         0.5s
Running "flutter packages get" in flutter_localizations...   0.5s
Running "flutter packages get" in project_root...            0.2s
Running "flutter packages get" in flutter_test...            0.5s
Running "flutter packages get" in main...                    0.3s
Running "flutter packages get" in flutter_tools...           0.6s
Running "flutter packages get" in fuchsia_remote_debug_protocol...  0.5s
Downloading lcov data for package:flutter...                 0.3s

Ran 'pub' 31 times and fetched coverage data in 15.6s.

C:\Users\Kevin\Documents\flutter\examples\catalog>flutter run
Updating flutter tool...
Running "flutter packages get" in catalog...
Because sample_catalog depends on flutter_driver any from sdk which depends on yaml 2.1.14, yaml 2.1.14 is required.
So, because sample_catalog depends on yaml 2.1.13, version solving failed.
pub get failed (1)
```

```
```

```
C:\Users\Kevin\Documents\flutter\examples\catalog>flutter doctor -v
[√] Flutter (Channel beta, v0.5.1, on Microsoft Windows [Version 10.0.17134.112], locale en-US)
    • Flutter version 0.5.1 at C:\Users\Kevin\flutter
    • Framework revision c7ea3ca377 (3 weeks ago), 2018-05-29 21:07:33 +0200
    • Engine revision 1ed25ca7b7
    • Dart version 2.0.0-dev.58.0.flutter-f981f09760

[√] Android toolchain - develop for Android devices (Android SDK 28.0.0)
    • Android SDK at C:\Users\Kevin\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)
    • All Android licenses accepted.

[√] Android Studio (version 3.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 25.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)

[!] VS Code, 64-bit edition (version 1.23.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

! Doctor found issues in 1 category.
```

**Comment by readytopark** [comment]
It looks like it needs a higher version of dart(latest one is 2.0.0-dev.63.0 from the dart dev channel). How can I possibly change the Dart version being referenced when running `flutter run`?

**Comment by zoechi** [comment]
You can't. The Dart SDK is baked into Flutter and a specific Flutter version is bound to a specific Dart version.
Why do you think you need a higher Dart version?
Alternatively using an older version of a dependency that is compatible with the used Dart version should work.

**Comment by DanTup** [comment]
`flutter run` will use its own embedded version of Dart. I think the problem here is running `flutter update-packages --force-upgrade` on the beta branch. This is going to update all of the packages used internally by flutter which may not be compatible.

I just did this:

- Checked out beta branch
- Ran `flutter update-packages` (without `--force-upgrade`)
- `flutter run` in `catalog`

And it all worked fine. @LaughingCabbage what's the reason for trying to force upgrade all the packages? (you didn't include the original error, only after the upgrade, which I think shouldn't be done).

**Comment by readytopark** [comment]
## Steps to reproduce
```
cd flutter/examples/flutter_gallery
flutter upgrade 
```

## Logs
```
...
Because flutter_gallery depends on path >=1.6.1 which requires SDK version >=2.0.0-dev.62.0 <2.0.0, version solving failed.
pub upgrade failed (1)
```

If I can't upgrade Dart, then I have to downgrade some of the dependencies(such as `path` package in my case) on my own which I believe there must exist some reasons that the developers added a specific version of dependency in pubspec.yaml.

@DanTup `flutter update-packages` didn't help either.

**Comment by zoechi** [comment]
@readytopark 

> I believe there must exist some reasons that the developers added a specific version of dependency

Isn't that the version that `--force-upgrade` wrote into `pubspec.yaml`?

**Comment by readytopark** [comment]
Now, it works... checking out to beta is the key..

README has to explicitly mention that the example project they are building should be on `beta` branch since `flutter` is default to beta setting.

Thanks! @LaughingCabbage

**Comment by wonsuyon** [comment]
wow solved, thanks @readytopark

**Comment by rupert2133** [comment]
@readytopark How did you fix it? how did you checkout to beta?

**Comment by zoechi** [comment]
@rupert2133 `flutter channel beta`

**Comment by frey1esm** [comment]
After `flutter channel beta` and `flutter update-packages --force-upgrade` I get:

`Because sample_catalog depends on flutter_driver any from sdk which depends on source_maps 0.10.6, source_maps 0.10.6 is required.
So, because sample_catalog depends on source_maps 0.10.5, version solving failed.`

**Comment by DanTup** [comment]
@frey1esm if you're just trying to use flutter, you shouldn't run `flutter update-packages --force-upgrade`. That command is for devs working on Flutter to update dependencies (which may require other code changes).

If you're trying to [upgrade the dependencies to submit a PR](https://github.com/flutter/flutter/wiki/Upgrading-dependencies) then you'll need to fix up any issues that it introduces (and in some cases, this may be blocked on external packages if there are version resolution issues).

I'll close this because I don't think there's an issue here, but do shout if you think this isn't the case.

**Comment by frey1esm** [comment]
@DanTup That command notwithstanding, here's what I did trying to load a demo:

0. `flutter channel`     (result:  *beta)
1. `git clone https://github.com/flutter/flutter.git`
2. `cd flutter`
3. `flutter update-packages`
4. `cd flutter/examples/flutter_gallery`
5. `flutter run`

result: 
```
Because flutter_gallery depends on flutter_driver any from sdk which depends on source_maps 0.10.6, source_maps 0.10.6 is required.
So, because flutter_gallery depends on source_maps 0.10.5, version solving failed.
```
Other examples such as `hello_world` and `catalog` (as stated by @LaughingCabbage ) have version conflicts as well.

```
Flutter 0.5.1 • channel beta • https://github.com/flutter/flutter.git
Framework • revision c7ea3ca377 (5 weeks ago) • 2018-05-29 21:07:33 +0200
Engine • revision 1ed25ca7b7
Tools • Dart 2.0.0-dev.58.0.flutter-f981f09760
```

**Comment by DanTup** [comment]
@frey1esm In your example steps, I think you may be running the beta version of `flutter` (from your `PATH`) inside a clone of the `master` channel?

**Comment by digitaljoni** [comment]
@frey1esm if you cloned flutter to a separate folder just for the examples... you would need to checkout the beta branch. so your steps should be:

1. flutter channel (result: *beta)
2. git clone https://github.com/flutter/flutter.git
3. cd flutter
4. **git checkout beta**
5. flutter update-packages
6. cd flutter/examples/flutter_gallery
7. flutter run

i hope this helps. ran into the same issue.

**Comment by 1and1get2** [comment]
flutter channel is beta (as default)
checkout beta branch
run `flutter update-packages` on  the root dir
run `flutter run` on `flutter/examples/flutter_gallery`

results:
```
Running "flutter packages get" in flutter_gallery...
Because flutter_gallery depends on flutter_driver any from sdk which depends on yaml 2.1.15, yaml 2.1.15 is required.
So, because flutter_gallery depends on yaml 2.1.13, version solving failed.
pub get failed (1)
```

flutter doctor -v
```
[✓] Flutter (Channel beta, v0.5.1, on Mac OS X 10.14 18A336e, locale en-NZ)
    • Flutter version 0.5.1 at /Library/Google/flutter
    • Framework revision c7ea3ca377 (8 weeks ago), 2018-05-29 21:07:33 +0200
    • Engine revision 1ed25ca7b7
    • Dart version 2.0.0-dev.58.0.flutter-f981f09760

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.1)
    • Android SDK at /Volumes/Persistence/Library/Google/Android/SDK
    • Android NDK at /Volumes/Persistence/Library/Google/Android/SDK/ndk-bundle
    • Platform android-28, build-tools 28.0.1
    • Java binary at: /Applications/ExternalApplications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/ExternalApplications/Xcode-beta.app/Contents/Developer
    • Xcode 10.0, Build version 10L177m
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio
    • Android Studio at /Applications/ExternalApplications/Android Studio 3.2 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b04)

[✓] Android Studio
    • Android Studio at /Applications/ExternalApplications/Android Studio 3.3 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b04)

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/ExternalApplications/Android Studio.app/Contents
    • Flutter plugin version 26.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Ultimate Edition (version 2018.1.6)
    • IntelliJ at /Applications/ExternalApplications/IntelliJ IDEA.app
    • Flutter plugin version 26.0.2
    • Dart plugin version 181.4892.1

[!] VS Code (version 1.25.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (1 available)
    • GT I9505 • 9c8b3912 • android-arm • Android 7.1.2 (API 25)

! Doctor found issues in 1 category.
```

**Comment by DanTup** [comment]
> Because flutter_gallery depends on flutter_driver any from sdk which depends on yaml 2.1.15, yaml 2.1.15 is required.

Something is wrong here - in the current beta (v0.5.1), `flutter_driver` also depends on `2.1.13` of `flutter_driver`:

https://github.com/flutter/flutter/blob/v0.5.1/packages/flutter_driver/pubspec.yaml#L65

This suggests something is not reset back to the beta version. You should look in the `.packages` file inside `flutter_gallery` to see where it's loading `flutter_driver` from and then figure out why it's trying to load a newer version. It may be due to a `.packages` or `pubspec.lock` file left over from when you were on a newer version or tried to upgrade. You should be able to delete all `.packages`/`pubspec.lock` (lock, not yaml!) files in your Flutter installation and then re-run `flutter update-packages` to fix.

Alternatively you could do a git clean (something like `git clean -df --dry-run` and then remove the `dry-run` when you've ensured it's only resetting/deleting the cache/lock files you'd expect). Of course, be careful where you're deleting files or running git clean commands from!

**Comment by 1and1get2** [comment]
I was unable to find the `.packages` file you mentioned, executing find on the root folder of the repo:
```
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   master  find ./ -name '.packages'                            ✔  1166  22:59:47
.//packages/flutter_tools/test/data/dart_dependencies_test/changed_sdk_location/.packages
.//packages/flutter_tools/test/data/dart_dependencies_test/syntax_error/.packages
.//packages/flutter_tools/test/data/dart_dependencies_test/asci_casing/.packages
.//packages/flutter_tools/test/data/dart_dependencies_test/good/.packages
.//packages/flutter_tools/test/data/dart_dependencies_test/bad_package/.packages
.//packages/flutter_tools/test/data/dart_dependencies_test/bad_path/.packages
```

`git clean -df --dry-run` gives nothing

**Comment by 1and1get2** [comment]
i didnt find any `.packages` files under(in this case) `flutter/examples/flutter_gallery` folder
and running `git clean -dxf --dry-run` only gives `.DS_Store` files

```
git clean -dxf --dry-run:
Would remove examples/.DS_Store
Would remove examples/flutter_gallery/.DS_Store
```

**Comment by 1and1get2** [comment]
i couldn't find any `.packages` files under any of the examples, and running `git clean -dxf --dry-run` only gives
```
Would remove examples/.DS_Store
Would remove examples/flutter_gallery/.DS_Store
```

**Comment by DanTup** [comment]
Can you try running these from your flutter checkout and post the full output of them all here (be sure to use the paths here, including the `../../`'s).

```
pwd
which flutter
git clean -dxf --dry-run
cat packages/flutter_driver/pubspec.yaml | grep yaml
cat examples/flutter_gallery/pubspec.yaml | grep yaml
bin/flutter --version
bin/flutter update-packages
cd examples/flutter_gallery/
../../bin/flutter run
```

**Comment by 1and1get2** [comment]
above lines works
```
Last login: Fri Jul 27 04:42:51 on console
 derek@Dereks-Mac  ~  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   beta v0.5.1  pwd                                      ✔  1191  04:43:21
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   beta v0.5.1  which flutter      ✔  1192  04:44:38
/Library/Google/flutter/bin/flutter
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   beta v0.5.1  git clean -dxf --dry-run
Would remove examples/.DS_Store
Would remove examples/flutter_gallery/.DS_Store
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   beta v0.5.1  cat packages/flutter_driver/pubspec.yaml | grep yaml
  yaml: 2.1.13 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   beta v0.5.1  cat examples/flutter_gallery/pubspec.yaml | grep yaml
  # Also update dev/benchmarks/complex_layout/pubspec.yaml
  yaml: 2.1.13 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   beta v0.5.1  bin/flutter --version
Downloading Dart SDK from Flutter engine 1ed25ca7b7e3e3e8047df050bba4174074c9b336...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 58.6M  100 58.6M    0     0  6750k      0  0:00:08  0:00:08 --:--:-- 9807k
Building flutter tool...
  ╔════════════════════════════════════════════════════════════════════════════╗
  ║ WARNING: your installation of Flutter is 57 days old.                      ║
  ║                                                                            ║
  ║ To update to the latest version, run "flutter upgrade".                    ║
  ╚════════════════════════════════════════════════════════════════════════════╝


Flutter 0.5.1 • channel beta • git@github.com:flutter/flutter.git
Framework • revision c7ea3ca377 (8 weeks ago) • 2018-05-29 21:07:33 +0200
Engine • revision 1ed25ca7b7
Tools • Dart 2.0.0-dev.58.0.flutter-f981f09760
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   beta v0.5.1  bin/flutter update-packages
Downloading Material fonts...                                1.7s
Downloading package sky_engine...                            1.1s
Downloading common tools...                                  1.8s
Downloading darwin-x64 tools...                              3.9s
Downloading android-arm-profile/darwin-x64 tools...          2.6s
Downloading android-arm-release/darwin-x64 tools...          1.7s
Downloading android-arm64-profile/darwin-x64 tools...        2.6s
Downloading android-arm64-release/darwin-x64 tools...        1.6s
Downloading android-x86 tools...                             2.9s
Downloading android-x64 tools...                             2.8s
Downloading android-arm tools...                             1.9s
Downloading android-arm-profile tools...                     2.7s
Downloading android-arm-release tools...                     2.4s
Downloading android-arm64 tools...                           2.0s
Downloading android-arm64-profile tools...                   2.0s
Downloading android-arm64-release tools...                   1.9s
Downloading ios tools...                                     4.5s
Downloading ios-profile tools...                             4.5s
Downloading ios-release tools...                             3.8s
Downloading Gradle Wrapper...                                0.8s
Running "flutter packages get" in bots...                   20.0s
Running "flutter packages get" in vitool...                 20.3s
Running "flutter packages get" in tools...                  20.8s
Running "flutter packages get" in ui...                     24.6s
Running "flutter packages get" in external_ui...            21.0s
Running "flutter packages get" in flavors...                21.0s
Running "flutter packages get" in platform_interaction...   21.1s
Running "flutter packages get" in channels...               22.1s
Running "flutter packages get" in manual_tests...           20.2s
Running "flutter packages get" in devicelab...              24.6s
Running "flutter packages get" in complex_layout...         22.6s
Running "flutter packages get" in microbenchmarks...        25.3s
Running "flutter packages get" in automated_tests...        20.2s
Running "flutter packages get" in hello_world...            20.5s
Running "flutter packages get" in layers...                 20.4s
Running "flutter packages get" in platform_view...           2.3s
Running "flutter packages get" in platform_channel_swift... 22.1s
Running "flutter packages get" in catalog...                21.6s
Running "flutter packages get" in flutter_view...            2.2s
Running "flutter packages get" in stocks...                 24.5s
Running "flutter packages get" in platform_channel...       21.9s
Running "flutter packages get" in flutter_gallery...        28.5s
Running "flutter packages get" in flutter_goldens...        22.4s
Running "flutter packages get" in flutter...                22.9s
Running "flutter packages get" in main...                    0.3s
Running "flutter packages get" in flutter_tools...           0.6s
Running "flutter packages get" in flutter_localizations...  20.9s
Running "flutter packages get" in fuchsia_remote_debug_protocol... 23.5s
Running "flutter packages get" in flutter_driver...         22.5s
Running "flutter packages get" in project_root...            0.2s
Running "flutter packages get" in flutter_test...           21.4s
Downloading lcov data for package:flutter...                 1.3s

Ran 'pub' 31 times and fetched coverage data in 583.7s.
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter/examples/flutter_gallery   beta v0.5.1  ../../bin/flutter run
Launching lib/main.dart on GT I9505 in debug mode...
Initializing gradle...                                       1.1s
Resolving dependencies...                                    1.0s
Running 'gradlew assembleDebug'...                           2.5s
Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...                 11.8s
I/FlutterActivityDelegate(16973): onResume setting current activity to this
Syncing files to device GT I9505...                          6.5s

🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
An Observatory debugger and profiler on GT I9505 is available at: http://127.0.0.1:8100/
For a more detailed help message, press "h". To quit, press "q".
Lost connection to device.
```
so it seems that even though the flutter version is exactly the same, they behaive slightly differently?
```
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter/examples/flutter_gallery   beta v0.5.1  flutter --version
Warning: the 'flutter' tool you are currently running is not the one from the current directory:
  running Flutter  : /Library/Google/flutter
  current directory: /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter
This can happen when you have multiple copies of flutter installed. Please check your system path to verify
that you're running the expected version (run 'flutter --version' to see which flutter is on your path).

Warning! The 'flutter' tool you are currently running is from a different Flutter
repository than the one last used by this package. The repository from which the
'flutter' tool is currently executing will be used instead.
  running Flutter tool: /Library/Google/flutter
  previous reference  : /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter
This can happen when you have multiple copies of flutter installed. Please check
your system path to verify that you are running the expected version (run
'flutter --version' to see which flutter is on your path).

Flutter 0.5.1 • channel beta • https://github.com/flutter/flutter.git
Framework • revision c7ea3ca377 (8 weeks ago) • 2018-05-29 21:07:33 +0200
Engine • revision 1ed25ca7b7
Tools • Dart 2.0.0-dev.58.0.flutter-f981f09760
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter/examples/flutter_gallery   beta v0.5.1  cd ../..
 derek@Dereks-Mac  /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter   beta v0.5.1  bin/flutter --version
Flutter 0.5.1 • channel beta • git@github.com:flutter/flutter.git
Framework • revision c7ea3ca377 (8 weeks ago) • 2018-05-29 21:07:33 +0200
Engine • revision 1ed25ca7b7
Tools • Dart 2.0.0-dev.58.0.flutter-f981f09760
```

**Comment by DanTup** [comment]
> It seems my `/Library/Google/flutter/bin/flutter` flutter instance won't create `.packages`

I'm not sure why that is (or whether `/Library` may have weird permissions), but I think the main issue is this:

> Warning: the 'flutter' tool you are currently running is not the one from the current directory:
>  running Flutter  : /Library/Google/flutter
>  current directory: /Volumes/Persistence/Data/Workspace/Flutter/Sample/flutter

It's not a good idea to have multiple flutter checkouts. When you're interacting with one, anything that runs `flutter` without a full path will get the one from `PATH`. I would recommend deleting one and having only the is in a user directory where there will not be any permissions issues.
